Sprint 1

sgp900 opened the pull request.

sgp900 committed:
sprint_1 messanger add code

gennady-bars commented on package.json:
```diff
@@ -0,0 +1,21 @@
+  "author": "",
+  "license": "MIT",
+  "devDependencies": {
+    "@anikin/parcel-transformer-handlebars": "^1.0.9",
```
Лучше задавать **фиксированные версии** зависимостям. Если нечаянно удалить `package-lock.json` и попытаться повторно установить зависимости с помощью команды `npm install`, то **npm** установит последние версии зависимостей, что может привести к ошибкам.

gennady-bars commented on package.json:
```diff
@@ -0,0 +1,21 @@
+    "@anikin/parcel-transformer-handlebars": "^1.0.9",
+    "@anikin/parcel-transformer-handlebars-precompile": "^1.0.1",
+    "@parcel/transformer-sass": "^2.7.0",
+    "express": "^4.18.2",
```
В `devDependencies` указывается то, что необходимо для сборки и разработки приложения, а в `dependencies` нужно указывать то, что необходимо непосредственно для работы приложения (express, uuid). Все линтеры, тесты и тд нужно только в `devDependencies` указывать, чтобы зря не раздувать кол-во пакетов в собранном для деплоя приложении.

gennady-bars commented on src/scss/_vars.scss:
```diff
@@ -0,0 +1,44 @@
+  --clr-white: hsl(0, 0%, 100%);
+  --clr-red: hsl(0, 100%, 50%);
+
+  --clr-lightest-gray: #fafafa;
```
* Отлично, что цвета вынесены в `css`-константы

gennady-bars commented on src/markup/partials/channel.hbs:
```diff
@@ -0,0 +1,14 @@
+<li class="channel">
+  <div class="channel__avatar"><img src="{{avatar}}" /></div>
```
Каждой картинке (тегу `img`) нужен информативный атрибут `alt` (например: имя картинки вставлять), так как при ошибке загрузки картинки на сайте подсветится **название** этой картинки (которое в `alt`). Еще это нужно для **плохо видящих** и `незрячих` людей - для них есть специальные средства, которые распознают текст в `alt` и проговаривают его. Подробнее тут https://habr.com/ru/post/341810/

gennady-bars commented on src/markup/partials/overlayFile.hbs:
```diff
@@ -0,0 +1,19 @@
+      Поменять
+    </button>
+  </div>
+</div>
```
В проекте в некоторых файлах есть ошибка `EOF`. Для `git` важно наличие **пустой строки** в конце файла. Подробнее тут: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log Можно в настройки IDE добавить `insert_final_newline = true`

gennady-bars commented on server.js:
```diff
@@ -0,0 +1,14 @@
+
+const express = require("express");
+const path = require("path");
+const PORT = 3000;
```
Лучше дать возможность поменять порт из консоли, если понадобится: `const PORT = process.env.PORT || 3000;`  . Если не задан порт, тогда будет `3000` по умолчанию.

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная


* `Readme` отлично оформлен (можно чуть меньше шрифт сделать)
* Проект задеплоен и отлично работает
* Отлично, что не забыли про `.gitignore`
* Отлично, что ограничиваете версию `node`
* Добавлен `ESLint` и `Stylelint`
* Отлично, что цвета вынесены в `css`-константы

но есть некоторые недочеты: 

* Вам нужно еще показать как-то страницы 404 и 500. Для этого нужно было на главной странице показать список ссылок, чтобы можно было по ним пройтись.  Нужно обернуть список с ссылками в тег `nav` по семантике, так как это навигация по сайту 
* По семантике нужно помещать весь главный контент страницы в тег `main`.
* Не могу запустить проект локально в режиме разработки на порту 3000 (как это в задании описано). Нужно сделать такой скрипт 
* В `devDependencies` указывается то, что необходимо для сборки и разработки приложения, а в `dependencies` нужно указывать то, что необходимо непосредственно для работы приложения (express, uuid). Все линтеры, тесты и тд нужно только в `devDependencies` указывать, чтобы зря не раздувать кол-во пакетов в собранном для деплоя приложении.
* По заданию просят сделать нормальную **файловую структуру**. Нужно страницы в папку `pages` сложить, а компоненты (кнопки, инпуты и тд) в папку `components`
* Каждой картинке (тегу `img`) нужен информативный атрибут `alt` (например: имя картинки вставлять), так как при ошибке загрузки картинки на сайте подсветится **название** этой картинки (которое в `alt`). Еще это нужно для **плохо видящих** и `незрячих` людей - для них есть специальные средства, которые распознают текст в `alt` и проговаривают его. Подробнее тут https://habr.com/ru/post/341810/
* У Вас все файлы в разном формате сделаны. Скрин https://disk.yandex.ru/i/Jv8cn-DPTEImSA . В `js` принято называть файлы в `camelCase`, а у Вас смесь `snake_case`, `kebeb-case` и `camelCase` и `kebab-snake_case`. Нужно привести все файлы и папки к единому стилю `camelCase`
* В проекте в некоторых файлах есть ошибка `EOF`. Для `git` важно наличие **пустой строки** в конце файла. Подробнее тут: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log Можно в настройки IDE добавить `insert_final_newline = true`


**Можно лучше**


* хорошей практикой является подключение `normalize.css`, чтобы **унифицировать** стилизацию под разные браузеры. Подробнее тут https://htmlacademy.ru/blog/html/about-normalize-css . И эти стили нужно получать **первее**всех других, чтобы переопределить изначально стили по умолчанию.



Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

sgp900 committed:
1. На основной странице добавлены ссылки на страницы 404 и 500;
2. Добавлены сематические теги nav, section, main;
3. Добавлен скрипт "develop" в package.json для запуска в режиме разработки на порту 3000;
4. Наведён порядок с зависимостями в dependencies осталось handlebars и express;
5. Изменена файловая структура, страницы перемещены в ./markup/pages, а компоненты в ./markup/components;
6. Всем тегам img прописан атрибут alt;
7. Все имена файлов переведены к формату camelCase;
8. Каждый файл имеет в конце имеет пустую строку;

sgp900 committed:
Для зависимостей теперь указаны фиксированные версии

Добавлена возможность указать порт в консоли

Убран отладочный console.log

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
